javascript/javascript1/week4/voice.js

```diff
@@ -0,0 +1,123 @@
+    result = new Date().toLocaleDateString("en-US", {
+      day: "numeric",
+      month: "long",
+      year: "numeric",
+    });
+  } else if (command.startsWith("what is")) {
+    return calcOperations(command);
+  } else if (command.includes(str9)) {
+    setTimeout(timer, 4000);
+    result = "Timer set for 4 seconds";
+  } else if (command.includes("how will")) {
+    result = guessAboutToday();
+  } else {
+    result = "error";
+  }
+  return result;
+}
+function saveName(string) {
+  const stringArray = string.split(" ");
+  const userName = stringArray[stringArray.length - 1];
+  return userName;
+}
+function addActivity(string) {
+  const startIndex = string.indexOf("add") + 4;
+  const endIndex = string.indexOf("to my");
+  const activity = string.substring(startIndex, endIndex).trim();
+  listArr.push(activity);
+  return activity;
+}
+
+function removeActivity(string) {
+  const startIndex = string.indexOf("remove") + 7;
+  const endIndex = string.indexOf("from my");
+  const activity = string.substring(startIndex, endIndex).trim();
+  const index = listArr.indexOf(activity);
+  listArr.splice(index, 1);
+  return `Removed ${activity} from your todo`;
+}
+function calcOperations(string) {
+  let output;
+  const stringArray = string.split(" ");
+  const number1 = Number(stringArray[2]);
+  const number2 = Number(stringArray[4]);
+  const operation = stringArray[3];
+  if (operation.includes("+")) {
+    output = number1 + number2;
+  } else if (operation.includes("-")) {
+    output = number1 - number2;
+  } else if (operation.includes("*")) {
+    output = number1 * number2;
+  } else if (operation.includes("/")) {
+    output = number1 / number2;
+  } else {
+    output = "Please give proper input";
+  }
+  return output.toString();
+}
+function timer() {
+  console.log("Timer Done");
+}
+function guessAboutToday() {
+  let todayLuck = "";
+  const words = {
+    positiveWords: ["great", "awesome", "lucky"],
+    negativeWords: ["challenging", "less fortunate", "not as expected"],
+  };
+  const combinedArray = words.positiveWords.concat(words.negativeWords);
+  let randomWord =
+    combinedArray[Math.floor(Math.random() * combinedArray.length)];
+
+  if (words.positiveWords.includes(randomWord)) {
+    todayLuck += `Wow!, This day will be a ${randomWord} day for you!! Enjoy!!!`;
+  } else {
+    todayLuck += `This day will be ${randomWord} for you, but remember every challenge brings opportunity. Stay positive and keep moving forward!`;
+  }
+
+  return todayLuck;
+}
+
+let listArr = [];
+//let str1;
+const str1 = "hello my name is kiruthiga";
+const str2 = "what is my name";
+const str3 = "add fishing to my todo";
+const str4 = "add singing in the shower to my todo";
+const str5 = "remove fishing from my todo";
+const str6 = "what is on my todo";
+const str7 = "what day is it today";
+const str8 = "what is 3 / 3";
+const str9 = "set timer for 4 seconds";
+const str10 = "how will the day be for me";
```

Comment:
Some of those variables are not used anymore. You can just delete them. Eg. `str8, 3, 4, 6, 10`, also the `// let str1;`

Reply:
I have removed unwanted declarations.